src/NameRegistry.sol

```diff
@@ -878,17 +883,64 @@ contract NameRegistry is
 
         // Perf: super.ownerOf is called instead of ownerOf since cancellation has no undesirable
         // side effects when expired and it saves some gas.
-        if (sender != super.ownerOf(tokenId) && sender != recoveryOf[tokenId]) revert Unauthorized();
+        if (sender != super.ownerOf(tokenId) && sender != registrationMetadataOf[tokenId].recovery) {
+            revert Unauthorized();
+        }
 
         // Check if there is a recovery to avoid emitting incorrect CancelRecovery events
-        if (recoveryClockOf[tokenId] == 0) revert NoRecovery();
+        if (recoveryMetadataOf[tokenId].recoveryTs == 0) revert NoRecovery();
 
         // Clear the recovery request so that it cannot be completed
-        delete recoveryClockOf[tokenId];
+        delete recoveryMetadataOf[tokenId].recoveryTs;
 
         emit CancelRecovery(sender, tokenId);
     }
 
+    /*//////////////////////////////////////////////////////////////
```

Comment:
supportive of your suggestion of removing view functions to save space since the mappings are public and can be accessed directly

Reply:
i removed the view functions and refactored the tests, saves like 300 bytes in deployed contract size